pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -187,20 +279,31 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
     [[nodiscard]] inline Threading threading() const { return threading_; }
 
     /**
-     *  @brief Returns a tuple containing the gate, generator, and matrix kernel
-     * maps respectively.
+     *  @brief Returns a tuple containing the gate, generator, and controlled
+     * matrix kernel maps respectively.
      */
-    [[nodiscard]] auto getSupportedKernels()
-        const & -> std::tuple<const GateKernelMap &, const GeneratorKernelMap &,
-                              const MatrixKernelMap &> {
-        return {getGateKernelMap(), getGeneratorKernelMap(),
-                getMatrixKernelMap()};
+    [[nodiscard]] auto getSupportedKernels() const & -> std::tuple<
+        const GateKernelMap &, const GeneratorKernelMap &,
+        const MatrixKernelMap &, const ControlledMatrixKernelMap &,
+        const ControlledGeneratorKernelMap &, const ControlledGateKernelMap &> {
+        return {getGateKernelMap(),
+                getGeneratorKernelMap(),
+                getMatrixKernelMap(),
+                getControlledMatrixKernelMap(),
+                getControlledGeneratorKernelMap(),
+                getControlledGateKernelMap()};
```

Comment:
If it is not a big trouble, would be great to have the same sequence ``gate, generator, matrix`` here and in other places.

Reply:
Will do, let me go over it again.